nixos/modules/services/networking/adguardhome.nix

```diff
@@ -12,23 +13,18 @@ let
     "--config /var/lib/AdGuardHome/AdGuardHome.yaml"
   ] ++ cfg.extraArgs);
 
-  configFile = pkgs.writeTextFile {
+  settings = if (cfg.settings != null)
+    then cfg.settings // {
+      http.address = "${cfg.host}:${toString cfg.port}";
+    } else null;
+
+  configFile = pkgs.writeTextFile rec {
     name = "AdGuardHome.yaml";
-    text = builtins.toJSON cfg.settings;
+    text = builtins.readFile (settingsFormat.generate name settings);
```

Comment:
I believe this leads to IFD. Maybe you could do something like this instead?

```nix
configFile = (settingsFormat.generate name settings).overrideAttrs (oldAttrs: {
  checkPhase = "${pkgs.adguardhome}/bin/adguardhome -c $out --check-config";
});
```

Reply:
Ahh nice didn't know that works.